submissions/Roophee/js-oop/app.js

```diff
@@ -0,0 +1,125 @@
+const CELL_WIDTH = 101,
+    CELL_HEIGHT = 83,
+    CELL_PADDING_BOTTOM = 25,
+    PLAYER_START_X = CELL_WIDTH * 2,
+    PLAYER_START_Y = CELL_HEIGHT * 5 - CELL_PADDING_BOTTOM,
+    ENEMY_START_X = -CELL_WIDTH,
+    FIRST_ENEMY_Y = CELL_HEIGHT * 1 - CELL_PADDING_BOTTOM,
+    SECOND_ENEMY_Y = CELL_HEIGHT * 2 - CELL_PADDING_BOTTOM,
+    THIRD_ENEMY_Y = CELL_HEIGHT * 3 - CELL_PADDING_BOTTOM,
+
+    MAX_SPEED = 200,
+    MIN_SPEED = 120,
+    CANVAS_AREA = {
+        width: CELL_WIDTH * 5,
+        height: CELL_HEIGHT * 6
+    };
+
+const Entity = function (x, y) {
+    this.x = x;
+    this.y = y;
+    this.speed = this.generateRandSpeed();
+}
+
+Entity.prototype.render = function () {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+
+Entity.prototype.generateRandSpeed = function () {
+    return Math.floor(Math.random() * (MAX_SPEED - MIN_SPEED)) + MIN_SPEED;
+}
+
+var Enemy = function (x, y) {
+    Entity.call(this, x, y);
+    this.sprite = 'images/enemy-bug.png';
+};
+
+Enemy.prototype = Object.create(Entity.prototype);
+
+Enemy.prototype.playerLose = function () {
+    player.resetGame();
+}
+
+Enemy.prototype.ateThePlayer = function () {
```

Comment:
A bit misleading naming. "ate" is a past tense for "eat", so this method reads "do something once have the player eaten". In fact this method decides if to eat the player at all, under certain conditions.